src/Microsoft.Dnx.Runtime/DependencyManagement/LockFile.cs

```diff
@@ -31,6 +31,35 @@ public bool IsValidForProject(Project project, out string message)
                 return false;
             }
 
+            // Verify the global search paths
+            message = $"The project search paths defined in {GlobalSettings.GlobalFileName} have changed";
+            GlobalSettings globalSettings;
+            if (GlobalSettings.TryGetGlobalSettings(ProjectRootResolver.ResolveRootDirectory(project.ProjectDirectory), out globalSettings))
```

Comment:
We already resolve the root directory. Why not pass it in? I'd also measure the performance difference this additional file read adds. We already parse it to figure out the nuget path.